src/tools/rbd_mirror/image_replayer/ReplayStatusFormatter.cc

```diff
@@ -152,11 +153,8 @@ bool ReplayStatusFormatter<I>::calculate_behind_master_or_send_update() {
 template <typename I>
 void ReplayStatusFormatter<I>::send_update_tag_cache(uint64_t master_tag_tid,
 						     uint64_t mirror_tag_tid) {
-
-  dout(20) << "master_tag_tid=" << master_tag_tid << ", mirror_tag_tid="
-	   << mirror_tag_tid << dendl;
-
-  if (master_tag_tid == mirror_tag_tid) {
+  if (master_tag_tid == 0 || master_tag_tid == mirror_tag_tid ||
```

Comment:
A ```tag_tid``` of zero is perfectly valid so it should still be loaded.

Reply:
I thought we start from tag_tid = 1?

Because I saw tag_tid of zero only for mirror_position, which I supposed was only for cases when it was initialized with `cls::journal::ObjectPosition()` and was not filled with a real position.

I saw a case when it tried to fetch tag_tid of zero and failed:
```
2018-02-06 15:53:46.602 7fcf89ffb700 -1 rbd::mirror::image_replayer::ReplayStatusFormatter: 0x7fcf2000d5e0 handle_update_tag_cache: error retrieving tag 0: (2) No such file or directory
```
Also, what `predecessor.tag_tid` is expected for this tag_tid of zero? I suppose 0, and then it would try to fetch again.

Anyway, I agree about `master_tag_tid <= mirror_tag_tid` condition, and then it will include `master_tag_tid == 0` too, so no need to explicitly test this. Updating.